powersimdata/input/input_data.py

```diff
@@ -96,20 +147,11 @@ def get_profile_version(self, grid_model, kind):
         if kind not in profile_kind:
             raise ValueError("kind must be one of %s" % " | ".join(profile_kind))
 
-        query = posixpath.join(
-            server_setup.DATA_ROOT_DIR,
-            server_setup.BASE_PROFILE_DIR,
-            grid_model,
-            kind + "_*",
-        )
-        stdin, stdout, stderr = self.data_access.execute_command("ls " + query)
-        if len(stderr.readlines()) != 0:
-            print("No %s profiles available." % kind)
-            version = []
-        else:
-            filename = [os.path.basename(line.rstrip()) for line in stdout.readlines()]
-            version = [f[f.rfind("_") + 1 : -4] for f in filename]
-        return version
+        resp = requests.get(f"{BASE_URL}/version.json")
```

Comment:
Do we want users to be able to use their own profiles that only exist locally? I'm thinking that in the previous Docker implementation, users could just add a new profile (like a new demand profile to test) to the `ScenarioData`directory and be able to use it, whereas now they'll either need to add it to our blob storage or create their own.

Reply:
I think we are just downloading a JSON files enclosing the profile version here. Otherwise the logic that we had before is still in place, i.e., we only download if there is not local copy. Also in ***plug***, I would say the user can have its own profile and use them in the various script via the `set_base_profile` method.

Reply:
What would get added to the ScenarioList if the user brought their own profile? "untracked"?

Reply:
Good point, there isn't a natural way to use a custom profile if we are strictly checking the version.json in blob storage. I think we'd need to do something like cache that file locally so it can be edited, or check both a local copy and the one in blob storage, etc. The profiles can still be put in the ScenarioData folder locally, but they would have a version that is considered valid to actually use it.

Reply:
> What would get added to the ScenarioList if the user brought their own profile? "untracked"?

Yeah it would get added to the user's ScenarioList. So their environment would be consistent, but couldn't be easily "merged" with another one (e.g. our server environment). I was going to suggest something like adding full references to files in the scenario list, but can't do much if the sources aren't public, e.g. in some other blob storage. Not sure if this is an issue at this point?

Reply:
I would say you can use your own profile if they are stored in ***ScenarioData/raw/GRID_MODEL/*** with the following drawbacks:
* they need to be named ***demand_SOMETHING.cs*** or hydro/solar/wind
* the prompt won't tell they are available
* if they are erased you won't be able to recover you scenario

Reply:
Right, only thing is the local version would have to match the values in blob storage, otherwise the call to `Create.set_base_profile` would fail. I think this comes down to having `InputData.get_profile_version` return local options as well so a user doesn't have to trick the software.

Reply:
You are right @jon-hagg, I forgot that we can only assign a profile if the version is in the returned list of available version:
```
    def set_base_profile(self, kind, version):
        """Sets demand profile.

        :param str kind: one of *'demand'*, *'hydro'*, *'solar'*, *'wind'*.
        :param str version: demand profile version.
        :raises ValueError: if no profiles are available or version is not available.
        """
        possible = self.get_base_profile(kind)
        if len(possible) == 0:
            raise ValueError("No %s profile available" % kind)
        elif version in possible:
            if kind == "demand":
                self.demand = version
            if kind == "hydro":
                self.hydro = version
            if kind == "solar":
                self.solar = version
            if kind == "wind":
                self.wind = version
        else:
            raise ValueError("Available %s profiles: %s" % (kind, " | ".join(possible)))
```

Reply:
It is not a problem specific to the blob storage, we had the same issue before with the profiles stored on the server. No?

Reply:
For client/server setup that's true. But interestingly in docker we are using `LocalDataAccess` to list the available profiles, so it would have found anything a user added to their local folder.

Reply:
Yeah, I don't think this is a feature that has been widely used, but I could see it being useful for collaborators going forward to test things out before releasing "official" profiles.

I was also thinking along the same lines as Jon's solution to have `get_profile_version` return both local and blob storage profiles, making it clear which are our profiles vs local-only and maybe some warnings about using unsupported profiles.

Reply:
That would be nice. How would we handle that in the client/server configuration? A user would be able to use its own profile, edit the ***ScenarioList*** file file with its profile version but none of the other user would be able to load this scenario because they don't have access to the profiles.

Reply:
Since `DataAccess` differentiates between configurations, we could add something like `data_access.get_local_profiles(kind)` that returns nothing for `SSHDataAccess` but returns any additional profiles for `LocalDataAccess`. We could call and append the results in `InputData`.

Depending on how fancy we want to get, we could also pass in the profiles already found in blob storage to both a) mark which profiles have been cached if a user wants to be careful of their internet usage and b) filter out only the extra profiles not in blob storage.

Reply:
Cool, I ended up thinking the same thing (my other ideas didn't pan out). Not sure I understand the part about passing in the profiles though?

Reply:
I was first thinking of passing in the profiles like `data_access.get_local_profiles(kind, blob_profiles)`, but looking at it now, there's no reason we can't do that filtering/marking at the `InputData` level instead of at the `DataAccess` level.

Reply:
Actually, sorry, I remembered what I was thinking. You would want to pass in the blob profiles so that in the `SSHDataAccess` instance, you could return the locally cached profiles but exclude anything not in the blob profiles list. In the `LocalDataAccess` instance, you would return both cached and local-only profiles.

Reply:
Ah thanks, that makes sense. Let me push a commit with how I'm thinking of implementing that.